src/openms/source/METADATA/ID/IdentificationDataConverter.cpp

```diff
@@ -201,39 +201,70 @@ namespace OpenMS
       progresslogger.setProgress(peptides_counter);
       const String& id = pep.getIdentifier();
       ID::ProcessingStepRef step_ref = id_to_step.at(id);
-      ID::InputFileRef file_ref;
-      if (!step_ref->input_file_refs.empty())
+      ID::InputFileRef inputfile;
+      if (!pep.getBaseName().empty())
       {
-        // @TODO: what if there's more than one input file?
-        file_ref = step_ref->input_file_refs[0];
+        const auto file_ref = id_data.registerInputFile(ID::InputFile(pep.getBaseName()));
```

Comment:
just: inputfile = id_data... ?

Reply:
can you elaborate?

Reply:
line https://github.com/hendrikweisser/OpenMS/pull/16/files/4346c892369fdd264045dba41ebb04d0b9f7c0f3#diff-eb57500eeb186958ca2acd00517194d8b96446a4b25002b0403d3c361268b378R208 seems redundant